### PR TITLE
feat: Add support for Python 3.10

### PR DESCRIPTION
**Pull request by Hanaasagi**
Fix the issue: https://github.com/saghul/pycares/issues/134

### CODE REVIEW

**Comment by Hanaasagi** [comment]
>  First-time contributors need a maintainer to approve running workflows. Learn more. 

Please approve running workflows, I need to runt the test in 3.10. Thanks!

**Comment by saghul** [comment]
Workflows approved

**Comment by Hanaasagi** [comment]
Emm, The CI failed because Python 3.10 is not found on ubuntu 20.04 when we used `actions/setup-python@v1`. I upgrade to `actions/setup-python@v2`. Test passed in my fork repo, https://github.com/Hanaasagi/pycares/runs/3230116789?check_suite_focus=true.

Could you approve the newest workflows again? Thanks.

**Comment by saghul** [comment]
Sorry for the delay, running now.

**Comment by saghul** [comment]
Note that there are other relevant workflows: https://github.com/saghul/pycares/tree/master/.github/workflows

It's fine to have 3.10 tested now, but once it hits stable I guess we'll want to release wheels.

**Comment by Hanaasagi** [comment]
Thanks for reminding. It looks like I need to add an `cp310-*` in the `CIBW_BUILD`.

https://github.com/saghul/pycares/blob/63dd17e94a33ce17213adda545204d04e912fb05/.github/workflows/release-wheels.yml#L27-L29

**Comment by saghul** [comment]
> Thanks for reminding. It looks like I need to add an `cp310-*` in the `CIBW_BUILD`.

Yeah. Would you like to add that in this PR? Otherwise looks like we are good to go!

**Comment by Hanaasagi** [comment]
> Yeah. Would you like to add that in this PR? Otherwise looks like we are good to go!

Yes. But I have encountered some problems, `libffi-dev` is missing only in the 3.10 when building wheels. I am trying to find a solution.

CI log: https://github.com/Hanaasagi/pycares/runs/3230989395?check_suite_focus=true

**Comment by Hanaasagi** [comment]
I change to the offical action `pypa/cibuildwheel@v2.0.1`. And cibuildwheel supports CPython 3.10 since v2.0.0, it needs to set `CIBW_PRERELEASE_PYTHONS` to enable building with pre-release versions of Python.

About `CIBW_PRERELEASE_PYTHONS`: https://cibuildwheel.readthedocs.io/en/stable/options/#prerelease-pythons.

File: .github/workflows/build-wheels.yml [hunk 1]
```diff
@@ -20,7 +20,11 @@ jobs:
...
-        uses: joerick/cibuildwheel@v1.11.0
+        uses: pypa/cibuildwheel@v2.0.1
+        with:
+          output-dir: ./wheelhouse
         env:
-          CIBW_BUILD: "cp36-* cp37-* cp38-* cp39-*"
+          CIBW_BUILD: "cp36-* cp37-* cp38-* cp39-* cp310-*"
           CIBW_ARCHS_LINUX: auto aarch64
+          CIBW_PRERELEASE_PYTHONS: True
+          CIBW_BEFORE_ALL_LINUX: yum install -y libffi-devel
```
Comment (saghul) [inline .github/workflows/build-wheels.yml:30]: Doesn't this run on Ubuntu?

Comment (Hanaasagi) [reply .github/workflows/build-wheels.yml:30]: cibuildwheel uses the manylinux docker image to build wheels. If we don’t use additional configuration, the default is manylinux2010, it's based on centos.

About why I add `CIBW_BEFORE_ALL_LINUX` to install `libffi`, please see this issue https://github.com/pypa/cibuildwheel/issues/783

Comment (Hanaasagi) [reply .github/workflows/build-wheels.yml:30]: manylinux2014 is based on unbutu.

**Comment by saghul** [comment]
Excellent work, cheers!

File: .github/workflows/build-wheels.yml (hunk 1 above)
Comment (saghul) [reply .github/workflows/build-wheels.yml:30]: Ack!

**Pull request merged**